BranchMetrics/ios-branch-deep-linking-attribution
# SDK-1831 update branch.json support

## Reference
SDK-1831 update branch.json support

## Summary
Removed deprecated json options for the old version of Apple Search Ads. 
Add `enableLogging` json option
Add `deferInitForPlugin` json option

New optional public API, `notifyNativeToInit`. When `deferInitForPlugin` is enabled in `branch.json` the SDK will cache the init until this method is called. The intended use case are platforms with runtimes that are a bit slow to startup, preventing clients from setting any early lifecycle configuration. The plugin should call this when it's registering the init callback. Note there is an assumption that this will be called before any methods that require the SDK be initialized.

## Motivation
Improve configuration support on platforms where the runtime startup is slower than the native code. For example, React Native and Unity.

## Type Of Change

- [] Bug fix (non-breaking change which fixes an issue)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing Instructions
Add a `branch.json` file to the iOS project. Make sure it is included in the `Copy Bundle Resources` build phase.
Start the app and note Branch init does not proceed. 
At some point after app launch, fire the `notifyNativeToInit` method. Which releases Branch init.

Note calls to APIs that expect the SDK to be initialized will enter an error state. The notify must come first.









cc @BranchMetrics/saas-sdk-devs for visibility.
